### PR TITLE
Change ProjectDescription model constructors to static functions

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/5714

### Short description 📝

We're making all constructors of `ProjectDescription` models internal in favor of public static constructors with the exception of:
- Top-level models like `Project` (any that needs to run `dumpIfNeeded`) -> this convention is inline with `PackageDescription` and imho makes sense

### How to test the changes locally 🧐

CI should pass

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/ProjectDescription/Arguments.swift [hunk 1]
```diff
@@ -4,17 +4,11 @@ import Foundation
 public struct Arguments: Equatable, Codable {
     public var environmentVariables: [String: EnvironmentVariable]
     public var launchArguments: [LaunchArgument]
-
-    public init(
+
```
Comment (danyf90) [inline Sources/ProjectDescription/Arguments.swift:7]:
```suggestion

```

File: Sources/ProjectDescription/EnvironmentVariable.swift [hunk 2]
```diff
@@ -10,12 +10,16 @@ public struct EnvironmentVariable: Equatable, Codable, Hashable, ExpressibleBySt
     public var isEnabled: Bool
 
     // MARK: - Init
-
-    public init(value: String, isEnabled: Bool) {
+    
+    init(value: String, isEnabled: Bool) {
         self.value = value
         self.isEnabled = isEnabled
     }
```
Comment (danyf90) [inline Sources/ProjectDescription/EnvironmentVariable.swift:17]: Can't we just remove this?

**Review by danyf90 — APPROVED** [review]

File: Sources/ProjectDescription/EnvironmentVariable.swift (hunk 2 above)
Comment (fortmarek) [reply Sources/ProjectDescription/EnvironmentVariable.swift:17]: No, if you define an `init(stringLiteral:)`, you will lose the init generated by the compiler

Comment (danyf90) [reply Sources/ProjectDescription/EnvironmentVariable.swift]: You are right! No strong opinion, I would define that in an extension, so we have the generated one for free.

**Pull request merged**